lib/components/user/monitored-trip/trip-basics-pane.js

```diff
@@ -107,6 +109,13 @@ class TripBasicsPane extends Component {
 
       return (
         <div>
+          {/* Prevent user from leaving when form has been changed, but
+          don't show it when they click submit or cancel. */}
+          <Prompt
+            when={this.props.dirty && !this.props.isSubmitting && !this.props.canceled}
+            message={`You have haven't saved your ${this.props.isCreating ? 'new ' : ''}trip yet. If you leave, ${this.props.isCreating ? 'it' : 'changes'} will be lost.`}
```

Comment:
Lines are too long. Please split up these long lines. A boolean variable could be created for the first one. The second one could also be calculated separately before returning this JSX. Also, please restructure these props at the very beginning of the render method.

Reply:
Thanks! I'm not entirely happy with the `unsavedChangesMessage` line break, but that's the only way I could think of that doesn't create a break in the popup.